Improve cloud simulation

Shimmen opened the pull request:
The glitch is fixed, and the cloud respawn logic is also much improved.

_**However**_, it would be nice to have the cloud speed be dependent of the plane speed as I explained. Does anyone have a suggestion how it should be done (@Vottivott @SabbePabbe @Lajsyl)?

Shimmen committed:
Add velocity & height-based spawning

Shimmen committed:
Fix transparent quad glitch.

It was of course a matter of depth masking...

Shimmen committed:
Implement proper cloud respawn while in PreJumpState

Lajsyl commented:
Would it be possible to have the speed of the clouds dependent on what FallState you're in? So when you change to FreeFall the world is also told to change the speed of the clouds?

Shimmen commented:
That is essentially what I'm thinking. But where should the direction come from? I.e. the direction the plane is going in.

Shimmen committed:
Implement airplane speed dependent cloud simulation.

 It's sort of hacked in, but it works for now. It also has some other shortcomings:

 (from TODOs)
  - Make sure clouds only spawn in the leading edge of the airplane direction, now it's possible that it spawn on the left side, and since the place goes right in high speed we will never see them.

  - Make sure there isn't a big blob of clouds when you encounter the first clouds spawned at the bottom (i.e. when fall state initially is changed to FreeFallingState here below)

Shimmen commented:
Det är inte perfekt, men det fungerar. Så jag mergar in det nu!
